packages/server/src/main/java/software/uncharted/terarium/hmiserver/service/data/DataMigrationESToPG.java

```diff
@@ -149,7 +149,8 @@ Pair<Integer, Integer> migrateFromEsToPg(final ElasticsearchService elasticServi
 					}
 				}
 
-				if (Objects.equals(lastId, lastPagesLastId) || assets.size() < PAGE_SIZE) {
+				if (Objects.equals(lastId, lastPagesLastId)
+						|| resp.hits().hits().size() < PAGE_SIZE) {
```

Comment:
Does there need to be any null safety here?
if resp, or either call to .hits fails this would blow up

Reply:
We already read the .hits().hits() directly. They seem to not ever be null even if no results come in.